### PR TITLE
Don't pass -enable-enterprise for fork prs

### PR DESCRIPTION
**Pull request by lkysow**
Fork PRs don't have access to the enterprise license secret.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -97,12 +103,18 @@ commands:
...
+                    ENABLE_ENTERPRISE=true
+                  fi
+
                   pkgs=$(go list ./... | circleci tests split --split-by=timings --timings-type=classname)
                   echo "Running $pkgs"
                   gotestsum --junitfile "$TEST_RESULTS/gotestsum-report.xml" -- $pkgs -p 1 -timeout 2h -failfast \
                       << parameters.additional-flags >> \
                       -enable-multi-cluster \
-                      -enable-enterprise \
+                      ${ENABLE_ENTERPRISE:+-enable-enterprise} \
```
Comment (lkysow) [inline .circleci/config.yml:117]: from https://stackoverflow.com/questions/42985611/how-to-conditionally-add-flags-to-shell-scripts

**Review by kschoche — APPROVED** [review]
Nice!

File: .circleci/config.yml [hunk 2]
```diff
@@ -65,6 +65,12 @@ commands:
               working_directory: test/acceptance/tests
               no_output_timeout: 2h
               command: |
+                # Enterprise tests can't run on fork PRs because they require
+                # a secret.
+                if [ -z "$CIRCLE_PR_NUMBER"]; then
```
Comment (ishustava) [inline .circleci/config.yml:70]:
```suggestion
                if [ -z "$CIRCLE_PR_NUMBER" ]; then
```
I tested it in #1018, and got `/bin/bash: line 2: [: missing `]'` [here](https://app.circleci.com/pipelines/github/hashicorp/consul-helm/3329/workflows/1a62922e-3eb2-4dcc-b92f-9c273e38534a/jobs/14607)

**Review by ishustava — APPROVED** [review]
Worked great! I tested in #1018, and everything ran perfectly without the enterprise tests. The only thing was that suggestion I left.

File: .circleci/config.yml (hunk 2 above)
Comment (lkysow) [reply .circleci/config.yml]: At least one of us tested this :D

**Pull request merged**